### DISCUSSION TITLE
Reversing the order of elements along some axis using Einops?

### DISCUSSION THREAD

**Discussion by ryunuri** [arogozhnikov/einops#203] [Q&A] [answered]
I think Einops operations are really comfortable and easy to use. So, I wanted to use einops for any kind of tensor manipulation.
However, I currently don't see a direct way to do a simple 'flip' or reversing the order of elements along some axis.

If I'm not mistaken, it seems that operation such as dividing an image into two halves and switching their place is not possible using only einops operations.

So, is there a way to do some flips using einops? If not, is there any plans for future implementations? Or should I just use some other functions provided by the framework I'm using?

**Comment by arogozhnikov** [comment, marked as answer]
Hi @terryryu, thank you for kind words.

There is currently no such operation as reversal of axes. 
I've had an experimental version of reversal for numpy, but seems like a too narrow case to include. Also, behavior would be inevtiably very different across frameworks (e.g. for numpy that's essentially a free operation, while torch will create a full copy).

For anyone seeking this feature I recommend upvoting this question.

**Comment by kkm000** [comment]
I reverse my axes all the time. Upvoted. :)

**Comment by genman** [comment]
I could use this.
